libs/providers/ofrep-web/src/lib/ofrep-web-provider.ts

```diff
@@ -0,0 +1,309 @@
+  Hook,
+  JsonValue,
+  Logger,
+  OpenFeatureError,
+  OpenFeatureEventEmitter,
+  Provider,
+  ProviderFatalError,
+  ResolutionDetails,
+  TypeMismatchError,
+} from '@openfeature/web-sdk';
+import { OfrepWebProviderOptions } from './model/ofrep-web-provider-options';
+import {
+  EvaluationFailureErrorCode,
+  EvaluationRequest,
+  EvaluationResponse,
+  handleEvaluationError,
+  isEvaluationFailureResponse,
+  isEvaluationSuccessResponse,
+  OFREPApi,
+  OFREPApiFetchError,
+  OFREPApiTooManyRequestsError,
+  OFREPApiUnauthorizedError,
+  OFREPForbiddenError,
+  RequestOptions,
+} from '@openfeature/ofrep-core';
+import { EvaluationStatus } from './model/evaluation-status';
+import {
+  InvalidContextError,
+  ParseError,
+  StandardResolutionReasons,
+  TargetingKeyMissingError,
+} from '@openfeature/core';
+import { isResolutionError, ResolutionError } from './model/resolutionError';
+
+export class OfrepWebProvider implements Provider {
+  DEFAULT_POLL_INTERVAL = 30000;
+
+  metadata = {
+    name: OfrepWebProvider.name,
+  };
+  readonly runsOn = 'client';
+  events = new OpenFeatureEventEmitter();
+  hooks?: Hook[] | undefined;
+
+  // logger is the Open Feature logger to use
+  private _logger?: Logger;
+  // _options is the options used to configure the provider.
+  private _options: OfrepWebProviderOptions;
+
+  private _ofrepAPI: OFREPApi;
+  private _etag: string | null;
+  private _pollingInterval: number;
+  private _retryPollingAfter: Date | undefined;
+  private _cache: { [key: string]: ResolutionDetails<FlagValue> | ResolutionError } = {};
+  private _context: EvaluationContext | undefined;
+  private _pollingIntervalId?: number;
+
+  constructor(options: OfrepWebProviderOptions, logger?: Logger) {
+    this._options = options;
+    this._logger = logger;
+    this._etag = null;
+    this._ofrepAPI = new OFREPApi(this._options.baseUrl);
+    this._pollingInterval = this._options.pollInterval ?? this.DEFAULT_POLL_INTERVAL;
+  }
+
+  resolveBooleanEvaluation(flagKey: string): ResolutionDetails<boolean> {
+    return this.evaluate(flagKey, 'boolean');
+  }
+  resolveStringEvaluation(flagKey: string): ResolutionDetails<string> {
+    return this.evaluate(flagKey, 'string');
+  }
+  resolveNumberEvaluation(flagKey: string): ResolutionDetails<number> {
+    return this.evaluate(flagKey, 'number');
+  }
+  resolveObjectEvaluation<T extends JsonValue>(flagKey: string): ResolutionDetails<T> {
+    return this.evaluate(flagKey, 'object');
+  }
+
+  /**
+   * onContextChange is called when the context changes, it will re-evaluate the flags with the new context
+   * and update the cache.
+   * @param oldContext - the old evaluation context
+   * @param newContext - the new evaluation context
+   */
+  async onContextChange(oldContext: EvaluationContext, newContext: EvaluationContext): Promise<void> {
+    try {
+      const now = new Date();
+      if (this._retryPollingAfter !== undefined && this._retryPollingAfter > now) {
```

Comment:
This is an interesting use case that we should discuss. I'm concerned about ignoring context change, but in this case, the server has already to try again at a later time. This could put us in a situation where the UI doesn't rerender components due to a context change.

Reply:
For now, I keep things stale but we may consider sending an error in that case so we will send the default values until the next evaluation.

Thinking about it, putting the provider in `ERROR` is probably better in that case.
WDYT?

Reply:
I've double-checked and we are already in `error` status when calling `onContextChange`, so from my point of view we are good, we will send the default value until we have reached the `Retry-after` date.

I've just spotted a bug, we were not storing the new context. Now it is stored and next time we will call the API (via polling we will have the new context set).